services/freecodecamp/freecodecamp-points.service.js

```diff
@@ -0,0 +1,69 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService, InvalidResponse, NotFound } from '../index.js'
+
+const schema = Joi.object({
+  entities: Joi.object({
+    user: Joi.object()
+      .required()
+      .pattern(/^\w+$/, {
+        points: Joi.number().allow(null).required(),
+      }),
+  }).optional(),
+}).required()
+
+/**
+ * This badge displays the total number of points a student has accumulated
+ * from completing challenges on freeCodeCamp.
+ */
+export default class FreeCodeCampPoints extends BaseJsonService {
+  static category = 'other'
+  static route = {
+    base: 'freecodecamp/points',
+    pattern: ':username',
+  }
+
+  static examples = [
+    {
+      title: 'freeCodeCamp points',
+      namedParams: { username: 'sethi' },
+      staticPreview: this.render({ points: 934 }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'points', color: 'info' }
+
+  static render({ points }) {
+    return { message: metric(points) }
+  }
+
+  static validate(response, username) {
+    const { entities } = response
+
+    if (entities === undefined)
+      throw new NotFound({ prettyMessage: 'profile not found' })
+
+    if (entities.user[username].points === null)
+      throw new InvalidResponse({ prettyMessage: 'private' })
+  }
+
+  async fetch({ username }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.freecodecamp.org/api/users/get-public-profile`,
+      options: {
+        qs: {
+          username,
+        },
+      },
+    })
+  }
+
+  async handle({ username }) {
+    const response = await this.fetch({ username })
+    this.constructor.validate(response, username)
+
+    const { points } = response.entities.user[username]
```

Comment:
Great idea to separate out the error throwing from `handle`, but let's combine all of this, including the current `validate` function, into a single `transform` function that's called from `handle` That would be much more consistent with our other service classes